brainiak/utils/utils.py

```diff
@@ -100,3 +102,127 @@ def fast_inv(a):
     except np.linalg.linalg.LinAlgError:
         logging.exception('Error from np.linalg.solve')
         raise
+
+
+def cov2corr(cov):
+    """Calculate the correlation matrix based on a
+        covariance matrix
+
+    Parameters
+    ----------
+
+    cov: 2D array
+
+    Returns
+    -------
+
+    corr: 2D array
+        correlation converted from the covarince matrix
+
+
+    """
+    assert cov.ndim == 2, 'covariance matrix should be 2D array'
+    inv_sd = 1 / np.sqrt(np.diag(cov))
+    corr = cov * np.outer(inv_sd, inv_sd)
+    return corr
+
+
+class read_design:
+    """ A class which has the ability of reading in design matrix in .1D file,
+        generated by AFNI's 3dDeconvolve.
+    """
+    def __init__(self, fname=None, include_orth=False, include_pols=False):
+        if fname is None:
+            # fname is the name of the file to read in the design matrix
+            self.design = np.zeros([0, 0])
+            self.n_col = 0
+            # number of columns (conditions) in the design matrix
+            self.column_types = np.ones(0)
+            self.n_basis = 0
+            self.n_stim = 0
+            self.n_orth = 0
+            self.StimLabels = []
+        else:
+            isAFNI = re.match(r'.+[.](1D|1d|txt)$', fname)
+            # We assume all AFNI 1D files have extension of 1D or 1d or txt
+            if isAFNI:
+                self.readAFNI(fname=fname)
+
+        self.include_orth = include_orth
+        self.include_pols = include_pols
+        self.cols_used = np.where(self.column_types == 1)[0]
+        if self.include_orth:
+            self.cols_used = np.sort(
+                np.append(self.cols_used, np.where(self.column_types == 0)[0]))
+        if self.include_pols:
+            self.cols_used = np.sort(np.append(
+                self.cols_used, np.where(self.column_types == -1)[0]))
+        self.design_used = self.design[:, self.cols_used]
+        if not self.include_pols:
+            # baseline is not included, then we add a column of all 1's
+            self.design_used = np.insert(self.design_used, 0, 1, axis=1)
+        self.n_TR = np.size(self.design_used, axis=0)
+
+    def readAFNI(self, fname):
+        # Read design file written by AFNI
+        self.n_basis = 0
+        self.n_stim = 0
+        self.n_orth = 0
+        self.StimLabels = []
+        self.design = np.loadtxt(fname, ndmin=2)
+        with open(fname) as f:
+            all_text = f.read()
+
+        find_n_column = re.compile(
```

Comment:
Actually I realized that there is a checking a few lines below, for whether the number of conditions read from the header matches the number of columns.